### PR TITLE
Adding documentation and changing website layout

### PR DESCRIPTION
**Pull request by Jagaskak**
Signed-off-by: Akshaya Jagannadharao <akshaya2@illinois.edu>

Website I'm trying to build should look like this one: https://jagaskak.github.io/mtt/

Lots of changes (I captured the more important ones here):
- Changed layout of website so it's easier to navigate
- Deleted duplicate _config.yml since it exists in the docs directory. 
- Updated Doxygen so that every plugin has short description of what it does
- Created pages folder for documentation
- Added placeholder files for documentation that is currently being created but will take some more time to review/add to
- Plugin and INI documentation created
- Updated pictures that were in the wiki

Things to look for:
- Information that is for the Perl Client only
- Out of date/inaccurate information
- Things that require better clarity
- Anything is helpful

### CODE REVIEW

File: docs/pages/plugins_docs.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- [TestDef.py](https://github.com/open-mpi/mtt/blob/master/pylib/System/TestDef.py)
+
+LoadClasses is called by TestDef and loads all plugins by going through the directories and loading plugins to memory.
+
+TestDef is passed to all other plugins so that they have access to the log, to global plugins (i.e. logger, modcmd, execmd, etc. that are saved to TestDef so they are accessible by other plugins) and to the options. This plugin is important for several reasons:
+- It is the center point for setting configurations for MTT
+- Sets the configuration for tests
+- Logs results
+- Loads plugins by called LoadClasses
+- Starts execution of tests by called one of the [Executor](https://github.com/open-mpi/mtt/tree/master/pylib/Tools/Executor) plugins. The [combinatorial](https://github.com/open-mpi/mtt/blob/master/pylib/Tools/Executor/combinatorial.py) plugin is currently the default plugin; however, the [sequential](https://github.com/open-mpi/mtt/blob/master/pylib/Tools/Executor/sequential.py) plugin can be set by using the ```--executor=sequential``` flag.
```
Comment (PeterGottesman) [inline docs/pages/plugins_docs.md]: The default executor is sequential:
https://github.com/open-mpi/mtt/blob/75cd103f305224ac995fad462057c1b34a5de9e6/pyclient/pymtt.py#L273-L277

Comment (Jagaskak) [reply docs/pages/plugins_docs.md]: Ok! Fixed it. Thanks for pointing that out.

File: docs/README.md [hunk 2]
```diff
@@ -1,224 +1,65 @@
...
-We have no illusions of MTT becoming the be-all/end-all tool for
-testing software -- we do want to keep it somewhat focused on the
-needs and requires of testing MPI implementations.  As such, the usage
-flow is somewhat structured towards that bias.
+# Overview
 
-It should be noted that the software has been mostly developed internally
-to the Open MPI project and will likely experience some growing pains
-while adjusting to a larger community.
+MTT is divided into multiple phases of execution to split up grabbing content, building content, running content, and reporting results (please refer to the [INI documentation](/mtt/docs/ini_docs.html) to learn more).
```
Comment (PeterGottesman) [inline docs/README.md]: INI documentation URL should be `/mtt/pages/ini_docs.html`

**Comment by PeterGottesman** [comment]
Akshaya, this looks really good. I haven't had much time to look over it, but what I have seen so far is good. I will try to look over/fact check over the weekend, but it is likely good to merge.

refs #671

**Pull request merged**